src/unitxt/operators.py

```diff
@@ -677,6 +680,74 @@ def process(self, stream: Stream, stream_name: str = None) -> Generator:
                 yield instance
 
 
+class ExtractFieldValues(MultiStreamOperator):
+    field: str
+    stream_name: str
+    overall_top_frequency_percent: Optional[int] = 100
+    min_frequency_percent: Optional[int] = 0
+    to_field: str
+    process_every_value: Optional[bool] = True
+
+    """
+    Extract the unique values of a field ('field') of a given stream ('stream_name') and store (the most frequent of) them
+    as a list in a new field ('to_field') in all streams.
+
+    More specifically, sort all the unique values encountered in field 'field' by decreasing order of frequency.
+    When 'overall_top_frequency_percent' is smaller than 100, trim the list from bottom, so that the total frequency of
+    the remaining values makes 'overall_top_frequency_percent' of the total number of rows in the stream.
+    When 'min_frequency_percent' is larger than 0, remove from the remaining list any value whose relative frequency makes
+    less than 'min_frequency_percent' of the total number of rows in the stream.
+
+    Examples:
+
+    ExtractFieldValues(stream_name="train", field="label", to_field="classes") - extracts all the unique values of
+    field "label" and stores in field "classes" of each and every record in all streams.
+
+    ExtractFieldValues(stream_name="train", field="labels", to_field="classes", process_every_value=True) -
+    in case that field 'labels' contains a list of values (and not a single value) - track the occurrences of all the possible
+    values in these lists, and report the most frequent values.
+    if process_every_value=False, track the most frequent whole lists, and report those (as a list of lists) in field 'to_field'
+
+    ExtractFieldValues(stream_name="train", field="label", to_field="classes",overall_top_frequency_percent=80) -
+    extracts the most frequent possible values of field 'label' that cover at least 80% of the samples,
+    and stores them in field 'classes' of all streams.
+
+    ExtractFieldValues(stream_name="train", field="label", to_field="classes",min_frequency_percent=5) -
+    extracts all possible values of field 'label' that cover, each, at least 5% of the instances.
+    Stores these values, sorted by decreasing order of frequency, in field 'classes' of each record in all streams.
+    """
+
+    def process(self, multi_stream: MultiStream) -> MultiStream:
+        stream = multi_stream[self.stream_name]
+        all_values = []
+        for instance in stream:
+            if (not isinstance(instance[self.field], list)) or (self.process_every_value == False):
+                all_values.append(
```

Comment:
Add a comment why we convert the list to a tuple (took me a while to understand).

I think this is the logic:
1. If process_every_value and value is not a list, throw an error.
2. If not process_every_value and is not a list, append element to the all_values list.
3. If not process_every_value and list,  convert to tuple, and append tuple to all_values list.
4. If process_every_value and list, add each element in list to the all_values list.

Reply:
Exactly! conversion is done only to allow the simple use of Counter, which would not accept lists hashables - for its entries.